lib/distribute_reads.rb

```diff
@@ -59,8 +59,19 @@ def self.lag(connection: nil)
       begin
         # makara doesn't send SHOW queries to replica, so we must force it
         Thread.current[:distribute_reads][:replica] = true
-        status = connection.exec_query("SHOW SLAVE STATUS").to_hash.first
-        status ? status["Seconds_Behind_Master"].to_f : 0.0
+
+        unless @using_aws_aurora
+          status = connection.exec_query("SHOW SLAVE STATUS").to_hash.first
+          return status["Seconds_Behind_Master"].to_f if status
+        end
+
+        # If SHOW SLAVE STATUS returns the empty set, check if using AWS Aurora
+        # Aurora MySQL does not use MySQL Binary Log File Position Based Replication method for its replication,
+        # which is why the SHOW SLAVE STATUS command does not show any information.
+        @using_aws_aurora ||= connection.exec_query("SHOW TABLES FROM mysql LIKE 'ro_replica_status'").to_hash.first
```

Comment:
Needs to use connection pool id in caching, since there can be multiple connections (some that use Aurora, and some that don't). Also, this line gets executed every time if `@using_aws_aurora` is false.